source/pages/Tokens/ImportToken.tsx

```diff
@@ -83,27 +79,29 @@ export const ImportToken: FC = () => {
   };
 
   const addToken = async (token: ITokenEthProps) => {
-    try {
-      const provider = new ethers.providers.JsonRpcProvider(activeNetwork.url);
+    setIsLoading(true);
 
-      const metadata = await getTokenStandardMetadata(
-        token.contractAddress,
-        accounts[activeAccount.id].address,
-        provider
+    const addTokenMethodResponse =
+      await controller.wallet.assets.evm.addEvmDefaultToken(
+        token,
+        activeAccount.address,
+        activeNetwork.url
       );
 
-      const balance = `${metadata.balance / 10 ** Number(token.decimals)}`;
-      const formattedBalance = lodash.floor(parseFloat(balance), 4);
+    if (
+      typeof addTokenMethodResponse === 'boolean' ||
```

Comment:
Maybe changing this to something like `isBoolean(addTokenMethodResponse) || isNil(addTokenMethodResponse)` it might look a little bit better (both fn are from lodash)